addons/l10n_it/migrations/0.7/post-map_carryover_to_new_report.py

```diff
@@ -0,0 +1,76 @@
+from odoo import api, SUPERUSER_ID, tools
+
+def migrate(cr, version):
+    env = api.Environment(cr, SUPERUSER_ID, {})
+    vat_report_id = env.ref('l10n_it.tax_report_vat').id
+    monthly_vat_report_id = env.ref('l10n_it.tax_monthly_report_vat').id
+
+    external_value_cols = [
+        col
+        for col in tools.table_columns(env.cr, 'account_report_external_value')
+        if col not in ['id', 'carryover_origin_report_line_id', 'target_report_expression_id']
+    ]
+
+    cr.execute(f"""
+        SELECT report.id AS report_id,
+               expression.id AS expression_id,
+               report_line.code,
+               external_value.target_report_expression_id,
+               external_value.carryover_origin_report_line_id,
+               {', '.join(f'external_value.{col}' for col in external_value_cols)}
+          FROM account_report AS report
+          JOIN account_report_line AS report_line ON report.id = report_line.report_id
+          JOIN account_report_expression AS expression ON report_line.id = expression.report_line_id
+                                                      AND expression.engine = 'external'
+     LEFT JOIN account_report_external_value AS external_value ON expression.id = external_value.target_report_expression_id
+         WHERE (
+                   report.id = %s
+                   AND external_value.company_id IS NOT NULL
+               )
+            OR report.id = %s
+      ORDER BY expression_id;
+    """, (vat_report_id, monthly_vat_report_id))
+    report_info = cr.fetchall()
+
+    code2expression_id = {
+        report_line_code: target_report_expression_id
+        for report_id, _, report_line_code, target_report_expression_id, *_ in report_info
+        if report_id == monthly_vat_report_id
```

Comment:
Isn't it impossible to have a target expression for the monthly report since it didn't exist?